src/commands/api/deployWorkspaceProjectApi.ts

```diff
@@ -0,0 +1,81 @@
+import { type ResourceGroup } from "@azure/arm-resources";
+import { ResourceGroupListStep, parseAzureResourceGroupId } from "@microsoft/vscode-azext-azureutils";
+import { createSubscriptionContext, subscriptionExperience, type IActionContext, type ISubscriptionActionContext, type ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { type AzureSubscription } from "@microsoft/vscode-azureresources-api";
+import { Uri, type WorkspaceFolder } from "vscode";
+import { ext } from "../../extensionVariables";
+import { getWorkspaceFolderFromPath } from "../../utils/workspaceUtils";
+import { deployWorkspaceProjectInternal, type DeployWorkspaceProjectInternalContext } from "../deployWorkspaceProject/deployWorkspaceProjectInternal";
+import { listCredentialsFromRegistry } from "../image/imageSource/containerRegistry/acr/listCredentialsFromRegistry";
+import type * as api from "./vscode-azurecontainerapps.api";
+
+export async function deployWorkspaceProjectApi(context: IActionContext, deployWorkspaceProjectOptions: api.DeployWorkspaceProjectOptionsContract): Promise<api.DeployWorkspaceProjectResults> {
+    const { resourceGroupId, rootPath, dockerfilePath, srcPath, suppressConfirmation, suppressContainerAppCreation, ignoreExistingDeploySettings, shouldSaveDeploySettings } = deployWorkspaceProjectOptions;
+
+    const subscription: AzureSubscription = await subscriptionExperience(context, ext.rgApiV2.resources.azureResourceTreeDataProvider, {
+        selectBySubscriptionId: getSubscriptionIdFromOptions(deployWorkspaceProjectOptions),
+        showLoadingPrompt: false
+    });
+    const subscriptionContext: ISubscriptionContext = createSubscriptionContext(subscription);
+
+    const rootFolder: WorkspaceFolder | undefined = rootPath ? getWorkspaceFolderFromPath(rootPath) : undefined;
+    const resourceGroup: ResourceGroup | undefined = resourceGroupId ? await getResourceGroupFromId({ ...context, ...subscriptionContext }, resourceGroupId) : undefined;
+
+    const deployWorkspaceProjectInternalContext: DeployWorkspaceProjectInternalContext = Object.assign(context, {
+        ...subscriptionContext,
+        subscription,
+        resourceGroup,
+        rootFolder,
+        srcPath: srcPath ? Uri.file(srcPath).fsPath : undefined,
+        dockerfilePath: dockerfilePath ? Uri.file(dockerfilePath).fsPath : undefined,
+        suppressConfirmation,
+        ignoreExistingDeploySettings,
+        shouldSaveDeploySettings: !!shouldSaveDeploySettings,
+    });
+
+    const deployWorkspaceProjectResultContext = await deployWorkspaceProjectInternal(deployWorkspaceProjectInternalContext, undefined, {
+        // Don't show activity log updates in ACA when another client extension calls into this API.
+        // Let each client decide how it wants to show its own activity log updates.
+        suppressActivity: true,
+        suppressConfirmation,
+        suppressContainerAppCreation,
```

Comment:
Let me know if you think it would be a good idea to default some of these values